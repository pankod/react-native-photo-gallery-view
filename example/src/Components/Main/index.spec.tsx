import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

jest.mock('@pankod/project-splash', () => 'true');

import Main from './index';
import { Text } from 'react-native';

describe('Main Component', () => {
	let wrapper: ShallowWrapper;

	beforeEach(() => {
		wrapper = shallow(<Main />);
	});

	it('renders correctly', () => {
		expect(wrapper).toMatchSnapshot()
	});

	it('initialize state correctly', () => {
		expect(wrapper.state().intro).toBeTruthy()
	});

	it('state check after timeout', () => {
		jest.runAllTimers();
		wrapper.update();
		expect(wrapper.state().intro).toBeFalsy()
	});

	it('render component view after splash timeout', () => {
		jest.runAllTimers();
		wrapper.update();
		expect(wrapper.find(Text)).toHaveLength(1);
	});
});
