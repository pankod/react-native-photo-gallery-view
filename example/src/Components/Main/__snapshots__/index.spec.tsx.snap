// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component renders correctly 1`] = `ShallowWrapper {}`;
